crates/graphql-composition/src/ingest_subgraph/schema_definitions/link.rs

```diff
@@ -124,17 +124,28 @@ fn ingest_grafbase_extension_from_link(
 ) {
     if link_url.url.scheme() == "file" {
         let Some(name) = r#as.or_else(|| {
-            link_url.url.to_file_path().ok().and_then(|path| {
-                let file_name = path.file_name()?.to_str()?;
-                match file_name {
-                    // This is going to be the name for locally built extension. In that case, the parent directory has a more descriptive name.
-                    "build" => path
-                        .parent()
-                        .and_then(|parent| parent.file_name()?.to_str())
-                        .map(|s| subgraphs.strings.intern(s)),
-                    _ => Some(subgraphs.strings.intern(file_name)),
-                }
-            })
+            let mut segments = link_url.url.path_segments()?;
```

Comment:
Does that work for file urls? I guess so, if the tests are ok

Reply:
yeah same thing